miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -239,91 +234,106 @@ public BoxUse getUse(long id) throws IOException {
 
       // get persisted version of new box contents before change
-      List<BoxableId> ids = box.getBoxables().values()
+      List<BoxableId> ids = box.getBoxPositions().values()
           .stream()
-          .map(b -> b.getId())
+          .map(b -> b.getBoxableId())
           .collect(Collectors.toList());
       Map<BoxableId, BoxableView> oldOccupants = boxStore.getBoxableViewsByIdList(ids)
           .stream()
-          .collect(Collectors.toMap(BoxableView::getId, b -> b));
+          .collect(Collectors.toMap(BoxableView::getId, Functions.identity()));
 
       // Process additions/moves
-      Set<BoxableId> handled = Sets.newHashSet();
-      for (Map.Entry<String, BoxableView> entry : box.getBoxables().entrySet()) {
-        BoxableView previousOccupant = original.getBoxable(entry.getKey());
-        BoxableView newOccupant = entry.getValue();
-        handled.add(newOccupant.getId());
+      Set<BoxPosition> movedWithinBox = Sets.newHashSet();
+      List<BoxableView> movedFromOtherBoxes = new ArrayList<>();
+      for (Map.Entry<String, BoxPosition> entry : box.getBoxPositions().entrySet()) {
+        BoxPosition managedPos = managed.getBoxPositions().get(entry.getKey());
+        BoxPosition newPos = entry.getValue();
 
-        if (previousOccupant != null && newOccupant.getId().equals(previousOccupant.getId())) {
+        if (managedPos != null && newPos.getBoxableId().equals(managedPos.getBoxableId())) {
           // Unchanged
           continue;
         }
         if (message.length() > 0) {
           message.append("\n");
         }
 
-        BoxableView oldOccupant = oldOccupants.get(newOccupant.getId());
+        BoxableView oldOccupant = oldOccupants.get(newPos.getBoxableId());
         if (oldOccupant.getBoxId() != null) {
           if (oldOccupant.getBoxId().longValue() == box.getId()) {
             // Moved within same box
             message.append(String.format("Relocated %s (%s) from %s to %s", oldOccupant.getAlias(), oldOccupant.getName(),
                 oldOccupant.getBoxPosition(), entry.getKey()));
+            movedWithinBox.add(newPos);
           } else {
             // Moved from a different box
             message.append(String.format("Moved %s (%s) from %s (%s) to %s", oldOccupant.getAlias(), oldOccupant.getName(),
                 oldOccupant.getBoxAlias(), oldOccupant.getBoxName(), entry.getKey()));
-
-            Box oldHome = boxStore.get(oldOccupant.getBoxId());
-            String oldHomeMessage = String.format("Moved %s (%s) to %s (%s)", oldOccupant.getAlias(), oldOccupant.getName(),
-                original.getAlias(), original.getName());
-            addBoxContentsChangeLog(oldHome, oldHomeMessage);
+            movedFromOtherBoxes.add(oldOccupant);
           }
-          boxStore.removeBoxableFromBox(oldOccupant);
         } else {
           message.append(String.format("Added %s (%s) to %s", oldOccupant.getAlias(), oldOccupant.getName(), entry.getKey()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Added %s (%s) to %s" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)